### PR TITLE
Only run ci on PRs and master branch - not on every branch

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://progress.opensuse.org/issues/80394

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: ci
 # yamllint disable-line rule:truthy
-on: [push, pull_request]
+on:
+  pull_request:
```
Comment (b10n1k) [inline .github/workflows/ci.yml:4]: IMO this should also have master branch. is there any reason to run the CI for PR on branches other than the master?

**Comment by okurz** [comment]
I don't understand the purpose of this PR. Why do you not want to run tests on non-master branch? E.g. one might want to run checks before creating a PR or we want to run checks for a "sle12" branch

File: .github/workflows/ci.yml (hunk 1 above)
Comment (okurz) [reply .github/workflows/ci.yml:4]: why would you want to exclude non-master branches? If there is for example a "sle12" branch, we should also run tests for that branch

Comment (b10n1k) [reply .github/workflows/ci.yml:4]: i see your point. My question is why then you need ci on push only for master then? i think i miss something!

**Comment by asdil12** [comment]
Then we should list the `sle12` branch maybe explicitly.
I just experienced that the testsuite runs already even on my fork before I even create a PR.
And there it fails for some reason, I wasn't able to identify: https://github.com/asdil12/os-autoinst-distri-opensuse/runs/1649557089?check_suite_focus=true
It worked fine when the CI ran for the PR from that very branch: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11689/checks

So if this reproducible, every time I create a PR, I would get an email that my branch broke the CI.
This PR should prevent that.

**Comment by asdil12** [comment]
Found the reason why this yaml issues were only detected on my branches:

From the Makefile:
```Makefile
.PHONY: test-yaml-valid
test-yaml-valid:
    @# Get list of changed yaml files. We only want to lint changed files for
    @# now because yamllint complains about a lot of existing files
    $(eval CHANGED_YAMLS=$(shell sh -c "git --no-pager diff --diff-filter=dr --name-only origin/master | grep '\(schedule\|test_data\)/.*\.y.\?ml$$'"))
    if test -n "$(CHANGED_YAMLS)"; then \
      export PERL5LIB=${PERL5LIB_} ; tools/test_yaml_valid $(CHANGED_YAMLS);\
      which yamllint >/dev/null 2>&1 || echo "Command 'yamllint' not found, can not execute YAML syntax checks";\
      yamllint -c .yamllint $(CHANGED_YAMLS);\
    else \
      echo "No yamls modified.";\
    fi
```

I wonder why this doesn't apply for PR checks as well. Maybe Github is simulating an already merged master for PR checks?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (b10n1k) [reply .github/workflows/ci.yml:4]: seems you had similar question yesterday and there were some answers. i will resolve this conversation.

**Comment by okurz** [comment]
> I wonder why this doesn't apply for PR checks as well. Maybe Github is simulating an already merged master for PR checks?

maybe

**Pull request closed without merging**